### PR TITLE
Don't add ellipsis when size is greater than number of words

### LINKED ISSUE #506: truncatewords - truncate indicator is always added, even if shorter
If i use `truncatewords` in one of my OC liquid templates, the 3 dots are always visible, even if the text is much shorter.
image

This happend to me, when i used it like this `truncatewords: 40`, but the 3 dots are also shown on a text with 20 words or something like that.
image

Changing the indicator to a custom one also doesn't change that behavior.

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #506

### CODE REVIEW

File: Fluid.Tests/StringFiltersTests.cs [hunk 1]
```diff
@@ -300,7 +300,9 @@ public void TruncateWithCustomEllipsis(string input, int size, string ellipsis,
         [InlineData("The cat came back the very next day", 4, "The cat came back...")]
         [InlineData("The cat came back the very next day", 1, "The...")]
         [InlineData("The cat came back the very next day", 0, "...")]
-        [InlineData("The    cat came  back", 10, "The    cat came  back...")]
+        [InlineData("The cat came back the very next day", 8, "The cat came back the very next day...")]
```
Comment (sebastienros) [inline Fluid.Tests/StringFiltersTests.cs:303]: Can you confirm this is the behavior in Shopify?

File: Fluid/Filters/StringFilters.cs [hunk 2]
```diff
@@ -299,7 +299,10 @@ public static ValueTask<FluidValue> TruncateWords(FluidValue input, FilterArgume
                 source = "";
             }
 
-            source += ellipsis;
+            if (size <= words)
```
Comment (sebastienros) [inline Fluid/Filters/StringFilters.cs:302]: Shouldn't this be done also in the truncate filter?

File: Fluid.Tests/StringFiltersTests.cs (hunk 1 above)
Comment (hishamco) [reply Fluid.Tests/StringFiltersTests.cs:303]: I just tried them in Liquid JS

File: Fluid/Filters/StringFilters.cs (hunk 2 above)
Comment (hishamco) [reply Fluid/Filters/StringFilters.cs:302]: Nope

File: Fluid.Tests/StringFiltersTests.cs (hunk 1 above)
Comment (sebastienros) [reply Fluid.Tests/StringFiltersTests.cs:303]: I know. LiquidJs is not bullet proof, I have files issues there when I found bugs.

Tests to add:
https://github.com/Shopify/liquid/blob/ff1c35b986451cca14d9da9b2ef772cbc9c65b24/test/integration/standard_filter_test.rb#L217-L234

And one that got removed I think and which was wrong, is to add multiple spaces, they should be reduced to a single single one, like a `String.Join(" ", String.Split(RemoveEmptyEntries))`

Comment (sebastienros) [reply Fluid.Tests/StringFiltersTests.cs:303]: I see the test wasn't removed, but the result it wrong.

Comment (hishamco) [reply Fluid.Tests/StringFiltersTests.cs:303]: I see, I will check the ruby code tomorrow, coz I feel sleepy ;)

**Pull request merged**